Mathlib/Algebra/Order/Monoid/Lemmas.lean

```diff
@@ -0,0 +1,1404 @@
+    [ContravariantClass α α (swap (· * ·)) (· ≤ ·)] (a : α) {b c : α} :
+    b * a ≤ c * a ↔ b ≤ c :=
+  rel_iff_cov α α (swap (· * ·)) (· ≤ ·) a
+
+end LE
+
+section LT
+
+variable [LT α]
+
+@[simp, to_additive]
+theorem mul_lt_mul_iff_left [CovariantClass α α (· * ·) (· < ·)]
+    [ContravariantClass α α (· * ·) (· < ·)] (a : α) {b c : α} :
+    a * b < a * c ↔ b < c :=
+  rel_iff_cov α α (· * ·) (· < ·) a
+
+@[simp, to_additive]
+theorem mul_lt_mul_iff_right [CovariantClass α α (swap (· * ·)) (· < ·)]
+    [ContravariantClass α α (swap (· * ·)) (· < ·)] (a : α) {b c : α} :
+    b * a < c * a ↔ b < c :=
+  rel_iff_cov α α (swap (· * ·)) (· < ·) a
+
+@[to_additive add_lt_add_left]
+theorem mul_lt_mul_left' [CovariantClass α α (· * ·) (· < ·)] {b c : α} (bc : b < c) (a : α) :
+    a * b < a * c :=
+  CovariantClass.elim _ bc
+
+@[to_additive lt_of_add_lt_add_left]
+theorem lt_of_mul_lt_mul_left' [ContravariantClass α α (· * ·) (· < ·)] {a b c : α}
+    (bc : a * b < a * c) :
+    b < c :=
+  ContravariantClass.elim _ bc
+
+@[to_additive add_lt_add_right]
+theorem mul_lt_mul_right' [i : CovariantClass α α (swap (· * ·)) (· < ·)] {b c : α} (bc : b < c)
+    (a : α) :
+    b * a < c * a :=
+  i.elim a bc
+
+@[to_additive lt_of_add_lt_add_right]
+theorem lt_of_mul_lt_mul_right' [i : ContravariantClass α α (swap (· * ·)) (· < ·)] {a b c : α}
+    (bc : b * a < c * a) :
+    b < c :=
+  i.elim a bc
+
+end LT
+
+section Preorder
+
+variable [Preorder α]
+
+@[to_additive]
+theorem mul_lt_mul_of_lt_of_lt [CovariantClass α α (· * ·) (· < ·)]
+    [CovariantClass α α (swap (· * ·)) (· < ·)]
+    {a b c d : α} (h₁ : a < b) (h₂ : c < d) : a * c < b * d :=
+  calc
+    a * c < a * d := mul_lt_mul_left' h₂ a
+    _ < b * d := mul_lt_mul_right' h₁ d
+
+alias add_lt_add_of_lt_of_lt ← add_lt_add
+
+@[to_additive]
+theorem mul_lt_mul_of_le_of_lt [CovariantClass α α (· * ·) (· < ·)]
+    [CovariantClass α α (swap (· * ·)) (· ≤ ·)] {a b c d : α} (h₁ : a ≤ b) (h₂ : c < d) :
+    a * c < b * d :=
+  (mul_le_mul_right' h₁ _).trans_lt (mul_lt_mul_left' h₂ b)
+
+@[to_additive]
+theorem mul_lt_mul_of_lt_of_le [CovariantClass α α (· * ·) (· ≤ ·)]
+    [CovariantClass α α (swap (· * ·)) (· < ·)] {a b c d : α} (h₁ : a < b) (h₂ : c ≤ d) :
+    a * c < b * d :=
+  (mul_le_mul_left' h₂ _).trans_lt (mul_lt_mul_right' h₁ d)
+
+/-- Only assumes left strict covariance. -/
+@[to_additive "Only assumes left strict covariance"]
+theorem Left.mul_lt_mul [CovariantClass α α (· * ·) (· < ·)]
+    [CovariantClass α α (swap (· * ·)) (· ≤ ·)] {a b c d : α} (h₁ : a < b) (h₂ : c < d) :
+    a * c < b * d :=
+  mul_lt_mul_of_le_of_lt h₁.le h₂
+#align left.mul_lt_mul Left.mul_lt_mul
+
+/-- Only assumes right strict covariance. -/
+@[to_additive "Only assumes right strict covariance"]
+theorem Right.mul_lt_mul [CovariantClass α α (· * ·) (· ≤ ·)]
+    [CovariantClass α α (swap (· * ·)) (· < ·)] {a b c d : α}
+    (h₁ : a < b) (h₂ : c < d) :
+    a * c < b * d :=
```

Comment:
```suggestion
    (h₁ : a < b) (h₂ : c < d) : a * c < b * d :=
```
We could save a line here like this (I've been doing this, giving no special preference for a new line after `:`), but it's not important. (And eventually an automatic reformatter may take care of all this.)

Reply:
Just to check, is the situation that mathlib is agnostic on newline after `:`?

Or is there a preference for newline after `:` but it's not worth enforcing it here because it's not provided by mathport output and there might be an automatic reformatter someday?

Or is there a preference for _no_ newline after `:`?

Reply:
Preference for no newline after `:`. (But slight preference for after the colon rather than in the midst of arguments or in the middle of the return type.)